packages/yoga/src/Icon/Icon.jsx

```diff
@@ -49,13 +58,38 @@ Icon.propTypes = {
   /** Vertical size of the SVG. Use it as one of
    * theme.spacing tokens (xxsmall, small, medium...) */
   height: oneOf([...commonSizes, instanceOf(Number)]),
+  /** Size for vertical and horizontal of the SVG.
+   * Use it as one of theme.spacing tokens (xxsmall, small, medium...) */
+  size: (props, propName, componentName) => {
+    const { size, width, height } = props;
+
+    if (size) {
+      checkPropTypes(
+        {
+          [propName]: oneOfType([oneOf(commonSizes), string, number]),
+        },
+        { size },
+        'prop',
+        componentName,
+      );
+
+      if (width || height) {
```

Comment:
Maybe this condition should come first, instead of nested conditionals:

```js
if (width || height) throw new Error(/* ... */);

checkPropTypes(/* ... */)
```